### PR TITLE
Feature/direct simularium out

### PR DESCRIPTION
**Pull request by meganrm**
Problem
=======
Want to be able to write directly to simularium file

Solution
========
Make simularium a format option 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cellpack/autopack/Compartment.py [hunk 1]
```diff
@@ -125,67 +125,72 @@ class Compartment(CompartmentList):
...
+        gname=None,
+        object_name=None,
+        object_filename=None,
+        isBox=False,
+        isOrthogonalBoundingBox=None,
+        stype="mesh",
+        radius=0.0,
+        height=0.0,
+        axis=[0, 1, 0],
+    ):
```
Comment (meganrm) [inline cellpack/autopack/Compartment.py:149]: removed kws and made args explicit

File: cellpack/autopack/IOutils.py [hunk 2]
```diff
@@ -567,180 +568,12 @@ def addCompartments(env, compdic, i, io_ingr):
...
-        else:
-            json.dump(
-                env.jsondic, fp, separators=(",", ":"), default=default
-            )  # ,indent=4, separators=(',', ': ')
-    print("recipe saved to ", setupfile)
+def save_as_simularium(env, setupfile, all_ingr_as_array):
+    autopack.helper.clear()
+    autopack.helper.init_scene_with_objects(all_ingr_as_array)
+    autopack.helper.writeToFile(None, f"{setupfile}_results", env.boundingBox)
+    webbrowser.open("https://simularium.allencell.org/viewer")
```
Comment (prismofeverything) [inline cellpack/autopack/IOutils.py:576]: Mentioned in the call, but might be nice to be able to just save to simularium directly, then another function to open it in the browser as well.

File: cellpack/autopack/IOutils.py [hunk 3]
```diff
@@ -567,180 +568,12 @@ def addCompartments(env, compdic, i, io_ingr):
                         # setup recipe
                     o.setInnerRecipe(rMatrix)
 
-                    # put here the export/import ?
 
-
-def save_asJson(env, setupfile, useXref=True, indent=True):
```
Comment (prismofeverything) [inline cellpack/autopack/IOutils.py:573]: Love to see it go. What does `mixed` refer to in the one we're going with below? We can probably just call it `save_json` or something.

File: cellpack/autopack/Compartment.py [hunk 4]
```diff
@@ -125,67 +125,72 @@ class Compartment(CompartmentList):
...
+        filename=None,
+        gname=None,
+        object_name=None,
+        object_filename=None,
+        isBox=False,
+        isOrthogonalBoundingBox=None,
+        stype="mesh",
+        radius=0.0,
+        height=0.0,
+        axis=[0, 1, 0],
```
Comment (prismofeverything) [inline cellpack/autopack/Compartment.py:148]: It's great to see what the args are here

File: cellpack/autopack/Compartment.py [hunk 5]
```diff
@@ -2556,57 +1980,27 @@ def BuildGrid_kevin(self, env, superFine=False):
...
-            print(
-                "%s surface pts, %d inside pts, %d tot grid pts, %d master grid"
-                % (
-                    len(self.surfacePoints),
-                    len(self.insidePoints),
-                    nbGridPoints,
-                    len(env.grid.masterGridPositions),
-                )
-            )
-
```
Comment (prismofeverything) [inline cellpack/autopack/Compartment.py:2605]: That's a lot of red! One of the reddest PR's I've seen : ) pretty glorious

**Review by prismofeverything — APPROVED** [review]
Looks great! The cleanup to Compartment is especially dramatic. Things are improving.

**Pull request merged**